src/main/python/janusgraph_python/serializer/RelationIdentifierSerializer.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018 JanusGraph Python Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONUtil
+from ..core.datatypes.RelationIdentifier import RelationIdentifier
+
+
+class RelationIdentifierSerializer(object):
+    """
+    This is serializer method being used to serialize RelationIdentifier object which is used by JanusGraph for edge ID
+    """
+
+    GRAPHSON_PREFIX = "janusgraph"
+    GRAPHSON_BASE_TYPE = "RelationIdentifier"
+
+    @classmethod
+    def dictify(cls, relationID, writer):
+        """ Serializes RelationIdentifier object.
+
+        Args:
+            relationID (RelationIdentifier): The RelationID to serialize.
+            writer:
+
+        Returns:
+            json
+        """
+
+        relationJSON = cls.__relationID_to_json(relationID)
+
+        serializedJSON = GraphSONUtil.typedValue(cls.GRAPHSON_BASE_TYPE, relationJSON, cls.GRAPHSON_PREFIX)
+
+        return serializedJSON
+
+    @classmethod
+    def __relationID_to_json(cls, relationID):
+        """ JSONify the string representation of RelationID
+
+        Args:
+            relationID (RelationIdentifier): The RelationID object which needs to be dictified.
```

Comment:
I don't think that you need to document internal / private methods like this. These comments even at more confusion here as they contradict each other. Here, it says `dictified`, but three lines above it says `JSONify` and then even goes on to say `the string representation` although the method seems to take a `RelationIdentifier` object and **not** its string representation.

Reply:
I'll remove the docs here. That way it will be even better :-)

Reply:
First I got an error indicating that it won't work with Python installed just for the current user:
```bash
ImportError: cannot import name 'abc' from 'importlib' (C:\Workspace\GitHub\janusgraph\janusgraph-python\tempENV\lib\importlib\__init__.py)
Using base prefix 'c:\\users\\florian.hockmann\\appdata\\local\\programs\\python\\python37'
New python executable in C:\Workspace\GitHub\janusgraph\janusgraph-python\tempENV\Scripts\python.exe
ERROR: The executable C:\Workspace\GitHub\janusgraph\janusgraph-python\tempENV\Scripts\python.exe is not functioning
ERROR: It thinks sys.prefix is 'c:\\workspace\\github\\janusgraph\\janusgraph-python' (should be 'c:\\workspace\\github\\janusgraph\\janusgraph-python\\tempenv')
ERROR: virtualenv is not compatible with this system or executable
Note: some Windows users have reported this error when they installed Python for "Only this user" or have multiple versions of Python installed. Copying the appropriate PythonXX.dll to the virtualenv Scripts/ directory may fix this problem.
Fatal Python error: initsite: Failed to import the site module
```

So, I uninstalled Python and installed it again for all users. This led to this error:

```
Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: 'C:\\Program Files\\Python37\\Lib\\site-packages\\virtualenv.py'
Consider using the `--user` option or check the permissions.
```
and a few more, but I guess they are caused by the problem with virtual env.

Since it complained about missing permissions, I tried it again as Admin, which mostly worked:

```
prerequisite Python is installed.
Using base prefix 'c:\\program files\\python37'
New python executable in C:\Workspace\GitHub\janusgraph\janusgraph-python\tempENV\Scripts\python.exe
Installing setuptools, pip, wheel...
done.
pybuilder requires Python '!=3.0,!=3.1,!=3.2,<3.7,>=2.6' but the running Python is 3.7.2
Installed PyBuilder
Installed Sphinx
Building Docs
Generated API Docs
Der Befehl "pyb_" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
Build HTML Docs
Building Library
Der Befehl "pyb_" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
JanusGraph-Python successfully build
Installed JanusGraph-Python
```

where the German sentence _"Der Befehl "pyb\_" ist entweder falsch geschrieben oder
konnte nicht gefunden werden."_ means something like: _"The command "pyb\_" is either not spelled correctly or could not be found"_

Maybe it would be better if contributors would install a recent version of `pip` and `virtualenv` themselves? Also I'm wondering about pybuilder complaining about Python 3.7. If Python 3.7 is not supported yet (which seems to be the case: pybuilder/pybuilder#606), then it should probably be added to `BUILDING.md`.

Reply:
Oh okay okay. Should we be adding the versions supported? Well I do travis CI on following versions, and hence the following versions are for sure supported:

- Python 3.4
- Python 3.5
- Python 3.6
- Python 3.6dev
- Python 3.6dev
- Python 3.7dev.

We don't test on 3.7 yet, as that is pretty new, and might not be stable, as seen from Pybuilder not supporting it as of yet.

Also, if you look as README.md, you can see that the supported Python versions are `3.4, 3.5 and 3.6`.

Reply:
The README.md currently states:

> newer versions than 3.6, it is expected to work across all Python with major version as 3.

and BUILDING.md says:

> though is expected to work against any Python 3.

If we know that it doesn't work with 3.7, then we should mention that instead of suggesting that all 3 versions should work.